packages/shared/src/components/tabs/TabContainer.spec.tsx

```diff
@@ -54,4 +72,46 @@ describe('tab container component', () => {
     expect(third).toBeInTheDocument();
     expect(third).toHaveStyle({ display: 'none' });
   });
+
+  describe('with URLs', () => {
+    jest.mock('next/router', () => ({
+      useRouter() {
+        return {
+          isFallback: false,
+        };
+      },
+    }));
+
+    beforeAll(() => {
+      const mockPathname = '/second';
+      mocked(useRouter).mockImplementation(
+        () =>
+          ({
+            pathname: mockPathname,
+            push: jest.fn(),
+          } as unknown as NextRouter),
+      );
+    });
+
+    it('should set the active tab based on the current path', async () => {
+      renderUrlComponent({ shouldMountInactive: true });
+
+      const active = await screen.findByText('Second');
+      const inactive1 = await screen.findByText('First');
+      const inactive2 = await screen.findByText('Third');
+
+      expect(active).toHaveClass('bg-theme-active');
+      expect(inactive1).not.toHaveClass('bg-theme-active');
+      expect(inactive2).not.toHaveClass('bg-theme-active');
+    });
+
+    it('should redirect to the given URL on tab click', async () => {
+      const router = useRouter();
+      renderUrlComponent({ shouldMountInactive: true });
+
+      const first = await screen.findByText('First');
+      fireEvent.click(first);
+      expect(router.push).toHaveBeenCalledWith('/first');
```

Comment:
for some reason this assertion fails, it looks like the mock is not registering the function call? Maybe someone else has some thoughts here, in the meantime I'll try to fix it as well 🙏

Reply:
See example here on how to mock useRouter https://github.com/dailydotdev/apps/blob/dfa97065f51a017db5babf2f8bda03570c6f5dda/packages/shared/src/components/SidebarRankProgress.spec.tsx#L34

Reply:
right, of course, thanks @capJavert 🙏